src/frontend/sharedComponents/Text.tsx

```diff
@@ -1,15 +1,82 @@
 import * as React from 'react';
-import {Text as RNText, TextProps} from 'react-native';
+import {Text as RNText, StyleProp, TextProps, TextStyle} from 'react-native';
 import {BLACK} from '../lib/styles';
 
+type fontVariant = 'primary' | 'body';
+
+/**
+ *
+ * Font, font sizing and font weight, should follow designs in figma.
+ * Generally all headers, labels, and button us Rubik (set `variant` to "primary").
+ * Paragraph and body text should use the device's System Font.
+ * Refer to this [design sytem](https://www.notion.so/digidem/Typography-a1600603854445ad82b8d67d4067be1a) for more info.
+ */
 export const Text = ({
   children,
   style,
+  variant = 'body',
   ...otherTextProps
-}: React.PropsWithChildren<TextProps>) => (
-  <RNText
-    style={[{fontFamily: 'Roboto', color: BLACK, fontSize: 16}, style]}
-    {...otherTextProps}>
-    {children}
-  </RNText>
-);
+}: React.PropsWithChildren<TextProps> & {variant?: fontVariant}) => {
+  const computedStyle = React.useMemo(
+    () => [
+      style,
+      {fontFamily: getFontFamily({styleObj: style, variant: variant})}, // Set the correct fontFamily based on fontWeight
+    ],
+    [style, variant],
+  );
+
+  return (
+    <RNText
+      style={[
+        {color: BLACK, fontSize: variant === 'primary' ? 16 : 14},
+        computedStyle,
+      ]}
+      {...otherTextProps}>
+      {children}
+    </RNText>
+  );
+};
+
+function getFontFamily({
```

Comment:
I have some concerns about this method.
The mapping assumes that fontWeight directly corresponds to specific font files, which might not always be accurate. Can't styles come from various sources (inline styles, stylesheets, theme providers)?
Also, merging styles and inspecting them on each render seems like a lot of overhead.

Reply:
So in react native there is only one available font, the system font (Roboto for Android). So if we were to set the font to anything else, typescript won't throw an error, BUT the phone would still use the system font.

So in order to use a font, we need to include the font files (or import it through expo - which technically speaking just imports the file for us).

So a each font has a different file for each font weight and italization. So we cannot just import one rubik file. if we had the one rubik file and did `{fontFamily: 'rubik', fontWeight:"bold"}` the font weight won't show up. it would look like normal rubik. What we have to do is do` {fontFamily: 'Rubik_700Bold'}`. The `fontWeight` property technically only works for the system font.

So what this function does is allows developers to use fontWeight normally. And based on the fontWeight, the correct `fontFamily` will be chosen. So if the developer sets the `fontWeight` to 700, then this `getFontFamily `would return `{fontFamily: 'Rubik_700Bold'}`. 

Unfortunately with custom fonts we can't just use `fontWeight` and expect the system to know which file to use

Reply:
Thanks for this thorough explanation. I still think the logic is a little complex and having either separate components or a centralized typography style sheet would be more maintainable in the future.